src/optimizer/optimizer.cpp

```diff
@@ -150,6 +151,23 @@ unique_ptr<LogicalOperator> Optimizer::Optimize(unique_ptr<LogicalOperator> plan
 		cse_optimizer.VisitOperator(*plan);
 	});
 
+	// Once we know the column lifetime, we have more information regarding
+	// what relations should be the build side/probe side.
+	RunOptimizer(OptimizerType::BUILD_SIDE_PROBE_SIDE, [&]() {
+		vector<ColumnBinding> updating_columns;
+		if (plan->type == LogicalOperatorType::LOGICAL_UPDATE) {
```

Comment:
Would it be possible to move this logic into the optimizer, and add a comment explaining what it does? I think I remember this improving update query that contain a join because it retains the original rowid ordering.